inst/pages/exercises.qmd

```diff
@@ -1741,6 +1741,228 @@ pheatmap(
+of that sample type on the Shannon diversity index is statistically
+significant. A common threshold for significance is 0.05.
+iii. R-squared: This value indicates the proportion of the variance in the
+Shannon diversity index that is explained by the sample type. A higher R-squared
+value indicates a better fit of the model.
+
+Now, let's check some linear modeling assumptions :
+  
+```{r}
+#| label: sub-checking-modeling-assumptions
+#| code-fold: true
+#| code-summary: "Show solution"
+#| eval: false
+
+# - Residual diagnostics to visually check for normality and homoscedasticity.
+# - Durbin-Watson Test for independence of residuals
+
+par(mfrow = c(2, 2))
+plot(model)
+dw_result <- dwtest(model)
+print(dw_result)
+```
+
+Here, 
+
+i. Residual diagnostics (normality and homoscedasticity):
+
+a. The Residuals vs Fitted plot provides insight into the relationship 
+between residuals and fitted values. Ideally, the residuals should be randomly
+scattered around the horizontal line (residuals = 0) without any discernible 
+pattern. This indicates that the model’s assumption of linearity and homoscedasticity
+(constant variance of residuals) is likely met. If you observe any systematic patterns
+(such as a funnel shape), it may suggest heteroscedasticity, meaning the variance of
+residuals is not constant.
+b. The Normal Q-Q plot assesses whether the residuals are normally distributed. In this
+plot, residuals should align closely with the diagonal line. Deviations from this line,
+especially at the tails, indicate that the residuals are not normally distributed, 
+which could violate the assumption of normality.
+c. The Scale-Location (or Spread-Location) plot also helps check the homoscedasticity 
+assumption. It plots the square root of standardized residuals against the fitted values.
+d. The Residuals vs Leverage plot helps identify influential data points (outliers) that 
+might disproportionately affect the model. Points outside the dashed lines (Cook’s 
+distance) are potential outliers that may have a strong influence on the model's 
+coefficients.
+
+ii. Durbin-Watson test checks for autocorrelation in the residuals. A value
+close to 2 indicates no autocorrelation. Values significantly different 
+from 2 suggest autocorrelation.
+
+In a nutshell, if all the assumptions hold (as indicated by the diagnostic plots
+and the statistical tests), you can be confident that the linear regression model
+is appropriate for the data, and the results can be interpreted reliably.
+
+5. Further, fit linear model using a continuous variable, say "coverage", where
+Shannon diversity is the response variable, and coverage is the predictor. 
+
+"Coverage" is a measure of how comprehensively the microbial diversity
+was sampled in each sample. This could refer to the sequencing depth (number of reads)
+or an index like Good’s Coverage that quantifies the completeness of sampling.
```

Comment:
What is the motivation for comparing Shannon and coverage? This is a technical comparison where some association would be expected. Do we have any biologically relevant comparison available?

Reply:
including this comparison can highlight samples that need further sequencing, improving the quality of our data. We risk misinterpreting samples with high diversity but low coverage, where the sequencing depth might not have been sufficient to reveal the full diversity. It shows room to reduce bias but you are right this is an indirect relevance here.

Reply:
Also, I am extremely sorry for the delay in my response, my classes have started and keep me occupied but I shall resolve these soon.